web-api/src/storage/elastic/filter.rs

```diff
@@ -14,37 +14,96 @@
 
 use itertools::Itertools;
 use serde_json::{json, Value};
-use xayn_web_api_shared::serde::JsonObject;
+use xayn_web_api_shared::serde::{json_array, JsonObject};
 
 use crate::{
     personalization::filter::{CombineOp, CompareOp, Filter},
     storage::KnnSearchParams,
 };
 
+const BOOL: &str = "bool";
 const FILTER: &str = "filter";
 const MINIMUM_SHOULD_MATCH: &str = "minimum_should_match";
+const RANGE: &str = "range";
 const SHOULD: &str = "should";
 const TERM: &str = "term";
 const TERMS: &str = "terms";
 
-fn extend_value(filter: &mut JsonObject, occurence: &'static str, clause: Value) {
+fn merge_range(clause: &mut Vec<Value>) {
```

Comment:
:wrench:  maybe indicate in the name that this can only be used for conjunctions but not disjunctions of clauses.